test/ConvertToPriceInfoBeta_test.cpp

```diff
@@ -0,0 +1,96 @@
+#include <gtest/gtest.h>
+#include "ConvertToPriceInfoBeta.h"
+
+TEST(ConvertToPriceInfoBeta_test, first_test){
+    PriceInfo price;
+    price.symbol = "BTCBBB";
+    price.bid_price =1;
+    price.bid_quantity =10;
+    price.ask_price =2;
+    price.ask_quantity=20;
+    
+    PriceInfoBeta pricebeta1;
+    pricebeta1.from="BTC";
+    pricebeta1.to="BBB";
+    pricebeta1.bid_price =1;
+    pricebeta1.bid_quantity=10;
+    
+    PriceInfoBeta pricebeta2;
+    pricebeta2.from="BBB";
+    pricebeta2.to="BTC";
+    pricebeta2.bid_price =2;
+    pricebeta2.bid_quantity=20;
+    ConvertToPriceInfoBeta input;
+    std::vector<PriceInfo> input_data = {price};
+    std::vector<PriceInfoBeta> expected_output = {pricebeta1, pricebeta2};
+    EXPECT_EQ(input.convert(input_data), expected_output);
+}
+
+
+TEST(ConvertToPriceInfoBeta_test, second_test){
+    PriceInfo price;
+    price.symbol = "BBBBTC";
+    price.bid_price =1;
+    price.bid_quantity =10;
+    price.ask_price =2;
+    price.ask_quantity=20;
+
+    PriceInfoBeta pricebeta1;
+    pricebeta1.from="BTC";
+    pricebeta1.to="BBB";
+    pricebeta1.bid_price =2;
+    pricebeta1.bid_quantity=20;
+    
+    PriceInfoBeta pricebeta2;
+    pricebeta2.from="BBB";
+    pricebeta2.to="BTC";
+    pricebeta2.bid_price =1;
+    pricebeta2.bid_quantity=10;
+    ConvertToPriceInfoBeta input;
+    
+    auto result_conver = input.convert(std::vector<PriceInfo>({price}));
+    auto output = std::vector<PriceInfoBeta>({pricebeta2, pricebeta1});
+
+    EXPECT_EQ(result_conver, output);
+}
+
+TEST(ConvertToPriceInfoBeta_test, third_test){
+    PriceInfo price1;
+    price1.symbol = "AAABBB";
+    price1.bid_price =1;
+    price1.bid_quantity =10;
+    price1.ask_price =2;
+    price1.ask_quantity=20;
+
+    PriceInfo price2;
+    price2.symbol = "BBBCCC";
+    price2.bid_price =1;
+    price2.bid_quantity =10;
+    price2.ask_price =2;
+    price2.ask_quantity=20;
+
+    ConvertToPriceInfoBeta input;
+    auto result_conver = input.convert(std::vector<PriceInfo>({price1, price2}));
+    auto output = std::vector<PriceInfoBeta>();
+    EXPECT_EQ(result_conver, output);
+}
+
+TEST(ConvertToPriceInfoBeta_test, 4th_test){
```

Comment:
please add another test with inputs :
```
{
"AAABTC"
"BBBCCC"
"CCCDDD"
}
```

and another one for empty input `{}`

Reply:
solved